packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -209,37 +208,33 @@ export default function ProxyConfiguration(
     [proxyState.activeProxy, proxyState.proxies, updateProxyState]
   )
 
-  // show/hide the enable switch
   useEffect(() => {
-    if (proxyState.enabled) {
-      setShowEnableSwitch(proxyState.proxies.length > 0)
-    } else {
-      setShowEnableSwitch(proxyState.proxies.length > 0)
+    let outdated = false
+
+    if (!configured) {
+      return
     }
-  }, [showEnableSwitch, proxyState.enabled, proxyState.proxies])
 
-  useEffect(() => {
-    let removeConnectivityListener = () => {}
-    const checkConnectivity = async () => {
-      if (configured) {
-        const connectivity = await BackendRemote.rpc.getConnectivity(accountId)
-        setConnectivityStatus(connectivity)
-        removeConnectivityListener = onDCEvent(
-          accountId,
-          'ConnectivityChanged',
-          () =>
-            debounceWithInit(async () => {
-              const connectivity =
-                await BackendRemote.rpc.getConnectivity(accountId)
-              setConnectivityStatus(connectivity)
-            }, 300)()
-        )
+    const update = async () => {
+      const connectivity = await BackendRemote.rpc.getConnectivity(accountId)
+
+      if (outdated) {
+        return
       }
+
+      setConnectivityStatus(connectivity)
     }
-    checkConnectivity()
-    return () => {
-      removeConnectivityListener()
-    }
+
+    const debouncedUpdate = debounce(update, 300)
+    debouncedUpdate()
+    debouncedUpdate.flush()
```

Comment:
Is there any advantage to call flush here immediately instead of passing immediate: true as 3rd parameter to the debounce function?

Reply:
Yes. `immediate` would make it so that if two `ConnectivityChanged` events fire rapidly, the underlying function would only get invoked after the first one, but not after the second one.

```
debouncedUpdate()
debouncedUpdate.flush()
```

is practically equivalent to just

```
update()
```

except that it also starts the debounce period.